src/hypervisor.c

```diff
@@ -107,6 +108,43 @@ cc_oci_expand_net_device_cmdline(struct cc_oci_config *config, guint index) {
 	return g_strdup("");
 }
 
+/*
+ *
+ *
+ */
+#define DPDK_CMDLINE_OBJ "memory-backend-file,id=dpdkmem,size=2048M,mem-path=/dev/hugepages,share=on,prealloc=on"
+#define DPDK_CMDLINE_NUMA "node,memdev=dpdkmem"
+#define DPDK_CMDLINE_CHAR "socket,id=char1,path=/run/openvswitch/ovd_%s"
+#define DPDK_CMDLINE_NETD "type=vhost-user,id=mynet1,chardev=char1,vhostforce"
+#define DPDK_CMDLINE_DEV  "virtio-net-pci,netdev=mynet1,mac=%s"
+
+static gchar *
+cc_oci_expand_ovsdpdk_chardev_params(struct cc_oci_net_if_cfg *if_cfg) {
+	struct cc_oci_net_ipv4_cfg *ipv4_cfg = NULL;
+	gchar *ovs_port_path = NULL;
+	guint index = 0;
+
+	//
+	// Need to append the interface's IP in order to identify the full
+	// OVS port path.  I don't expect that there'd be multiple IPs
+	// associated wit hthe interface, but to be safe, walking through the
+	// list and double checking that the file exists on the system.
+	//
+
+	for (index=0; index<g_slist_length(if_cfg->ipv4_addrs); index++) {
+		ipv4_cfg = (struct cc_oci_net_ipv4_cfg *)
+			g_slist_nth_data(if_cfg->ipv4_addrs, index);
+		ovs_port_path = g_strdup_printf(OVS_PORT_PATH, ipv4_cfg->ip_address);
+
+		if (g_file_test (ovs_port_path, G_FILE_TEST_EXISTS)) {
```

Comment:
Yeah -- ouch.  Fixed these up, and reworked so I don't have to allocate as much..